resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -289,4 +278,92 @@ public String toString() {
             }
         }
     }
+
+    private static final class Entries {
+        private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+        // Access to this field is guarded by the ReadWriteLock.
+        //
+        // It's important that this List will never be modified itself but just replaced when an new entry is added,
+        // or removed.
+        private List<DefaultDnsCacheEntry> entries;
+
+        Entries(DefaultDnsCacheEntry entry) {
+            entries = Collections.unmodifiableList(Collections.singletonList(entry));
+        }
+
+        List<DefaultDnsCacheEntry> get() {
+            Lock readLock = lock.readLock();
+            readLock.lock();
```

Comment:
is the read lock necessary?

Reply:
Yes as we read the field... of we update it we use the write lock